From fe10fdc27a61a42a7789d357bd5b831523bb8cd3 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Thu, 17 Dec 2015 19:32:45 +0800
Subject: [PATCH 0982/1050] drm/amd/dal: change in MST connector detection

Only report connector as connected when edid available on
it. This will make sink available on connector in all
cases.

It also guarantees that we safe if any hotplug notifications
come.

Change-Id: I8fe7579fa3934fe20e2256764b01d8c1245c5b20
Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
index 3750a0d..4f585c6 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -79,6 +79,14 @@ dm_dp_mst_detect(struct drm_connector *connector, bool force)
 		aconnector->edid = NULL;
 	}
 
+	/*
+	 * we do not want to make this connector connected until we have edid on
+	 * it
+	 */
+	if (status == connector_status_connected &&
+		!aconnector->port->cached_edid)
+		status = connector_status_disconnected;
+
 	return status;
 }
 
-- 
1.9.1

